python_modules/dagster/dagster/_core/storage/event_log/base.py

```diff
@@ -361,6 +361,26 @@ def get_materialization_count_by_partition(
     ) -> Mapping[AssetKey, Mapping[str, int]]:
         pass
 
+    @abstractmethod
+    def get_partitions(self, partitions_def_name: str) -> Sequence[str]:
+        """Get the list of partition keys for a partition definition."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def has_partition(self, partitions_def_name: str, partition_key: str) -> bool:
+        """Check if a partition exists."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def add_partitions(self, partitions_def_name: str, partition_keys: Sequence[str]) -> None:
```

Comment:
I think we should be consistent on add_mutable_partitions vs. add_partitions at the different layers